monitor/launch/launcher.go

```diff
@@ -75,42 +75,52 @@ func (nm *NodeMonitor) GetPid() int {
 	return nm.GetProcess().Pid
 }
 
-// Run starts the node monitor.
-func (nm *NodeMonitor) Run(sockPath, bpfRoot string) {
-	nm.SetTarget(targetName)
-	for {
-		os.Remove(sockPath)
-		if err := syscall.Mkfifo(sockPath, 0600); err != nil {
-			log.WithError(err).Fatalf("Unable to create named pipe %s", sockPath)
-			time.Sleep(time.Duration(5) * time.Second)
-		}
+func (nm *NodeMonitor) run(sockPath, bpfRoot string) error {
+	os.Remove(sockPath)
+	if err := syscall.Mkfifo(sockPath, 0600); err != nil {
+		return fmt.Errorf("Unable to create named pipe %s: %s", sockPath, err)
+	}
```

Comment:
We shouldn't return in the EEXISTS case, otherwise after the first time we return from `run` we can never enter it again. Alternatively, add a deferred delete here since the monitor exits on any error (https://github.com/cilium/cilium/blob/master/monitor/monitor.go#L105-L106) and it won't run again until after this create occurs again.

Reply:
There is an `os.Remove()` above the `Mkfifo()`

Reply:
oh, there is... nevermind then